lib/livebook/session.ex

```diff
@@ -1363,6 +1386,11 @@ defmodule Livebook.Session do
     state
   end
 
+  defp after_operation(state, _prev_state, {:put_secret, _client_id, _secret}) do
+    if Runtime.connected?(state.data.runtime), do: set_runtime_secrets(state, state.data.runtime)
```

Comment:
I was thinking we can pass a list of secrets as an argument, so we only put that new secret, rather than sending all. It doesn't matter much, but we may as well do it :)

```suggestion
    if Runtime.connected?(state.data.runtime), do: set_runtime_secrets(state, [secret])
```

Also, we don't need to pass runtime as a separate argument, we can always access it form state.